server/models/helpers/SearchHelper.ts

```diff
@@ -354,36 +352,51 @@ export default class SearchHelper {
       });
     }
 
-    if (!options.includeArchived) {
-      where[Op.and].push({
-        archivedAt: {
-          [Op.eq]: null,
-        },
-      });
-    }
-
-    if (options.includeDrafts && model instanceof User) {
-      where[Op.and].push({
-        [Op.or]: [
+    const statusQuery = [];
+    if (options.statusFilter && options.statusFilter.includes("published")) {
+      statusQuery.push({
+        [Op.and]: [
           {
             publishedAt: {
               [Op.ne]: null,
             },
+            archivedAt: {
+              [Op.eq]: null,
+            },
           },
+        ],
+      });
+    }
+
+    if (options.statusFilter && options.statusFilter.includes("draft")) {
```

Comment:
```suggestion
    if (options.statusFilter?.includes("draft")) {
```

Collapse these clauses